added a live region that announces the selected item

ginader opened the pull request:
This is the one accessibility enhancement that works without any localization efforts.
I would love to also add the announcement that says to use the cursor keys and how many results there are but that would mean to add localization and this would add a huge maintenance overhead.
Updated: changed hiddena11y property to visually-hidden class

Commit:
add styles to hide only visually

Commit:
add a live region that announces the selected item

Commit:
changed hiddena11y property to visually-hidden class

LeaVerou commented:
Thanks Dirk. Did you see my other message?
Do we have to use an extra element? Won’t it work to just toggle these attributes on the current list item?
(btw you can update PRs, no need for new ones to make a modification)

ginader commented:
Hi Lea, yes I did see your other message and commented inline here: https://github.com/LeaVerou/awesomplete/pull/11#discussion_r25091213

Commit:
removed the setStatus helper and moved the updating of the aria-live region directly into the goto method

ginader commented:
I updated the pull request and answered you questions [here](https://github.com/LeaVerou/awesomplete/pull/11#discussion_r25125604) and [here](https://github.com/LeaVerou/awesomplete/pull/11#discussion_r25125500)

LeaVerou commented on awesomplete.css:
```diff
@@ -1,4 +1,14 @@
+	padding: 0;
+	position: absolute;
+	width: 1px;
+}
```
Sorry, I hadn’t realized I was commenting on the wrong PR. 
This element doesn’t get focus, so a lot of the concerns described in the article don’t make sense here.
I don’t want to add current verbosity for potential future changes. 
`padding: 0` and `border:0` aren’t needed in any way. `margin: -1px` also shouldn’t be needed. The clip syntax needs to use commas, this is the IE < 7 clip syntax, and we don’t care about IE < 7.

ginader replied:
You are right. For this use case some of the properties are not needed. I updated the CSS accordingly.

Commit:
removed border and padding and updated clip syntax

Border Margin and Padding are not needed in this case and Awesomeplete does not support IE7 so the clip syntax can be to standard with commas

LeaVerou commented:
If we're using `clip(0,0,0,0)` then why do we need a width and height or overflow?

ginader commented:
I haven't been testing using `clip(0,0,0,0)` exclusively across your supported Browsers. Would you say it's supported robustly across the Browsers you're supporting?

LeaVerou commented:
Sorry, I meant `clip: rect(0,0,0,0)`. Yes, it is supported but not sure if it’s ok with screen readers…

ginader commented:
From my experience and also [backed here](https://developer.yahoo.com/blogs/ydn/clip-hidden-content-better-accessibility-53456.html) by my former colleagues at Yahoo is that content hidden using `clip: rect(0,0,0,0)` is solidly accessible to assistive technologies. All the other additional methods are to handle behavior in legacy browsers. As you are already not supporting those I'd save we're save to go "light"

Commit:
removed all leftover legacy visually hiding methods

visually-hidden now uses exclusively clip: rect(0, 0, 0, 0);

LeaVerou commented:
Um, for `clip` to apply you need `position: absolute` too…

(sorry to keep going back and forth, I’m honestly not trying to avoid merging this; I care about it being accessible very much)

Commit:
readded position:absolute to make sure clip is working reliably

ginader commented:
I appreciate you making sure your code and UX is as good as possible. The fact that I care about the same thing is the reason why this pull request exists :-)

LeaVerou commented:
Finally merged, thanks!!

ginader commented:
Yay! :-)
On Feb 27, 2015 4:07 PM, "Lea Verou" notifications@github.com wrote:

> Finally merged, thanks!!
> 
> —
> Reply to this email directly or view it on GitHub
> https://github.com/LeaVerou/awesomplete/pull/26#issuecomment-76496233.
